gdb/riscv-tdep.c

```diff
@@ -578,9 +578,28 @@ riscv_print_register_formatted (struct ui_file *file, struct frame_info *frame,
 
       size = register_size (gdbarch, regnum);
       get_formatted_print_options (&opts, 'x');
+
+      char csize;
+      switch (size) {
+          case 1:
+            csize = 'b';
+            break;
+          case 2:
+            csize = 'h';
+            break;
+          case 4:
+            csize = 'w';
+            break;
+          case 8:
+            csize = 'g';
+            break;
+          default:
+            csize = 0;
```

Comment:
Can you error() here? Otherwise you'll just end up with an error inside print_scalar_formatted, which will be harder to debug.